Common/Helpers/BlobStorageManager.cs

```diff
@@ -36,31 +36,7 @@ public BlobStorageHelper(string connectionString, string containerName)
         /// </returns>
         public async Task<CloudBlobContainer> BuildBlobContainerAsync()
         {
-            if (string.IsNullOrEmpty(_connectionString))
-            {
-                throw new ArgumentException(
-                    "connectionString is a null reference or empty string.",
-                    "connectionString");
-            }
-
-            if (object.ReferenceEquals(_containerName, null))
-            {
-                throw new ArgumentNullException(_containerName);
-            }
-
-            CloudStorageAccount storageAccount;
-            if (!CloudStorageAccount.TryParse(_connectionString, out storageAccount))
-            {
-                throw new ArgumentException(
-                    "connectionString is not a valid Cloud Storage Account connection string.", "connectionString");
-            }
-
-            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-
-            CloudBlobContainer container = blobClient.GetContainerReference(_containerName);
-            await container.CreateIfNotExistsAsync();
-
-            return container;
+            return await _blobContainerProvider.GetCloudBlobContainerAsync();
```

Comment:
do we have to return CloudBlobContainer? Can we hide the implementation in this class by moving the methods that run on CloudBlobContainer in here?